samplePDF/samplePDFFDBase.cpp

```diff
@@ -1381,46 +1384,46 @@ void samplePDFFDBase::SetupNuOscillator() {
 #endif
         }
       } else {
-	int InitFlav = _BAD_INT_;
-	int FinalFlav = _BAD_INT_;
-	
-	if (std::abs(MCSamples[iSample].nutype) == 1) {
-	  InitFlav = NuOscillator::kElectron;
-	} else if (std::abs(MCSamples[iSample].nutype) == 2) {
-	  InitFlav = NuOscillator::kMuon;
-	} else if (std::abs(MCSamples[iSample].nutype) == 3) {
-	  InitFlav = NuOscillator::kTau;
-	}
-	
-	if (std::abs(MCSamples[iSample].oscnutype) == 1) {
-	  FinalFlav = NuOscillator::kElectron;
-	} else if (std::abs(MCSamples[iSample].oscnutype) == 2) {
-	  FinalFlav = NuOscillator::kMuon;
-	} else if (std::abs(MCSamples[iSample].oscnutype) == 3) {
-	  FinalFlav = NuOscillator::kTau;
-	}
-	
-	if (InitFlav == _BAD_INT_ || FinalFlav == _BAD_INT_) {
-	  MACH3LOG_ERROR("Something has gone wrong in the mapping between MCSamples[iSample].nutype and the enum used within NuOscillator");
-	  MACH3LOG_ERROR("MCSamples[iSample].nutype: {}", MCSamples[iSample].nutype);
-	  MACH3LOG_ERROR("InitFlav: {}", InitFlav);
-	  MACH3LOG_ERROR("MCSamples[iSample].oscnutype: {}", MCSamples[iSample].oscnutype);
-	  MACH3LOG_ERROR("FinalFlav: {}", FinalFlav);
-	  throw MaCh3Exception(__FILE__, __LINE__);
-	}
-	
-	//Assuming that if the generated neutrino is antineutrino, the detected flavour will also be antineutrino
-	if (MCSamples[iSample].nutype<0) {
-	  InitFlav *= -1;
-	  FinalFlav *= -1;
-	}
-	if (MCSamples[iSample].rw_truecz.size() > 0) { //Can only happen if truecz has been initialised within the experiment specific code
-	  //Atmospherics
-	  MCSamples[iSample].osc_w_pointer[iEvent] = NuOscProbCalcers[iSample]->ReturnWeightPointer(InitFlav,FinalFlav,FLOAT_T(*(MCSamples[iSample].rw_etru[iEvent])),FLOAT_T(*(MCSamples[iSample].rw_truecz[iEvent])));
-	} else {
-	  //Beam
-	  MCSamples[iSample].osc_w_pointer[iEvent] = NuOscProbCalcers[iSample]->ReturnWeightPointer(InitFlav,FinalFlav,FLOAT_T(*(MCSamples[iSample].rw_etru[iEvent])));
-	}
+        int InitFlav = _BAD_INT_;
+        int FinalFlav = _BAD_INT_;
+
+        if (std::abs((*MCSamples[iSample].nupdgUnosc[iEvent])) == NuPDG::kNue) {
+          InitFlav = NuOscillator::kElectron;
+        } else if (std::abs((*MCSamples[iSample].nupdgUnosc[iEvent])) == NuPDG::kNumu) {
+          InitFlav = NuOscillator::kMuon;
+        } else if (std::abs((*MCSamples[iSample].nupdgUnosc[iEvent])) == NuPDG::kNutau) {
+          InitFlav = NuOscillator::kTau;
+        }
+
+        if (std::abs((*MCSamples[iSample].nupdg[iEvent])) == NuPDG::kNue) {
+          FinalFlav = NuOscillator::kElectron;
+        } else if (std::abs((*MCSamples[iSample].nupdg[iEvent])) == NuPDG::kNumu) {
+          FinalFlav = NuOscillator::kMuon;
+        } else if (std::abs((*MCSamples[iSample].nupdg[iEvent])) == NuPDG::kNutau) {
+          FinalFlav = NuOscillator::kTau;
+        }
+
```

Comment:
I would simplfy this into
```
InitFlav = Func((*MCSamples[iSample].nupdgUnosc[iEvent])))
FinalFlav  = Func(*MCSamples[iSample].nupdg[iEvent])))
```

Reply:
I am nitpicky becasue once we start support sterile this will be terrific

Reply:
yeah that's fair, I'll change this. Don't want to have 3+N if statements!